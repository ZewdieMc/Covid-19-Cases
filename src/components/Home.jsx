import { useOutletContext } from 'react-router-dom';

const Home = () => {
  const { cases } = useOutletContext();
  return (
    <div>
      <h1>
        Home
        {' '}
        { cases.length }
      </h1>
    </div>
  );
};
export default Home;
