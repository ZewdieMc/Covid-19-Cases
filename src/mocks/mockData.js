export default [
  {
    country: 'Afghanistan',
    countryInfo: {
      _id: 4,
      iso2: 'AF',
      iso3: 'AFG',
      lat: 33,
      long: 65,
      flag: 'https://disease.sh/assets/img/flags/af.png',
    },
    cases: 214276,
    deaths: 7896,
    recovered: 193355,
    critical: 45,
  },
  {
    country: 'Albania',
    countryInfo: {
      _id: 8,
      iso2: 'AL',
      iso3: 'ALB',
      lat: 41,
      long: 20,
      flag: 'https://disease.sh/assets/img/flags/al.png',
    },
    cases: 334726,
    deaths: 3602,
    recovered: 329428,
    critical: 0,
  },
  {
    country: 'Algeria',
    countryInfo: {
      _id: 12,
      iso2: 'DZ',
      iso3: 'DZA',
      lat: 28,
      long: 3,
      flag: 'https://disease.sh/assets/img/flags/dz.png',
    },
    cases: 271688,
    deaths: 6881,
    recovered: 182943,
    critical: 0,
  },
  {
    country: 'Andorra',
    countryInfo: {
      _id: 20,
      iso2: 'AD',
      iso3: 'AND',
      lat: 42.5,
      long: 1.6,
      flag: 'https://disease.sh/assets/img/flags/ad.png',
    },
    cases: 47945,
    deaths: 165,
    recovered: 47563,
    critical: 14,
  },
  {
    country: 'Angola',
    countryInfo: {
      _id: 24,
      iso2: 'AO',
      iso3: 'AGO',
      lat: -12.5,
      long: 18.5,
      flag: 'https://disease.sh/assets/img/flags/ao.png',
    },
    cases: 105384,
    deaths: 1934,
    recovered: 103419,
    critical: 0,
  },
  {
    country: 'Anguilla',
    countryInfo: {
      _id: 660,
      iso2: 'AI',
      iso3: 'AIA',
      lat: 18.25,
      long: -63.1667,
      flag: 'https://disease.sh/assets/img/flags/ai.png',
    },
    cases: 3904,
    deaths: 12,
    recovered: 3879,
    critical: 4,
  },
];
