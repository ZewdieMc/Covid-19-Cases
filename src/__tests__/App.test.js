import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderWithProviders from '../utils/test-utils';
import LayOut from '../components/CovidLayOut';
import Home from '../components/Home';
import setupStore from '../redux/store';
import mockData from '../mocks/mockData';

describe('Covid Cases snapshot', () => {
  test('Test for Layout', () => {
    renderWithProviders(
      <Provider store={setupStore({})}>
        <LayOut />
      </Provider>,
    );
    const cases = document.querySelector('.cases');
    expect(cases).toMatchSnapshot();
  });
});

describe('Test COVID cases before fetching API', () => {
  test('Test for Loading... state', () => {
    renderWithProviders(<LayOut />);
    expect(screen.queryByText(/Loading.../i)).toBeInTheDocument();
    expect(screen.queryByText(/Confirmed/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/Deaths/i)).not.toBeInTheDocument();
  });
});

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useOutletContext: () => mockData,
}));

describe('Test COVID cases after fetching API', () => {
  test('Test if cases are rendered after fetching', async () => {
    renderWithProviders(<Home />);
    expect(screen.queryByText(/No Results Found/i)).toBeInTheDocument();
    expect(screen.queryByText(/Loading.../i)).not.toBeInTheDocument();
  });
});
