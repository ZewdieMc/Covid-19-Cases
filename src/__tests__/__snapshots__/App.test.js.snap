// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Covid Cases snapshot Test for Layout 1`] = `null`;
